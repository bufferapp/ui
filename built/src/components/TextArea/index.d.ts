export { default } from './TextArea';
//# sourceMappingURL=index.d.ts.map