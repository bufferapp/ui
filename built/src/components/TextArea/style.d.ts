export declare const Container: any;
export declare const StyledTextArea: any;
export declare const StyledLabel: any;
//# sourceMappingURL=style.d.ts.map