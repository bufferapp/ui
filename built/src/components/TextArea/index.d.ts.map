{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/TextArea/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,YAAY,CAAA"}