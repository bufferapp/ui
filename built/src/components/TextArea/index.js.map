{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/components/TextArea/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,YAAY,CAAA","sourcesContent":["export { default } from './TextArea'\n"]}