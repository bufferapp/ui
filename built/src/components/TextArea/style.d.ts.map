{"version":3,"file":"style.d.ts","sourceRoot":"","sources":["../../../../src/components/TextArea/style.ts"],"names":[],"mappings":"AAeA,eAAO,MAAM,SAAS,KAIrB,CAAA;AAwBD,eAAO,MAAM,cAAc,KAgC1B,CAAA;AAED,eAAO,MAAM,WAAW,KAGvB,CAAA"}