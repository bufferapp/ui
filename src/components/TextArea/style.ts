import styled from 'styled-components'
import {
  redDark,
  gray,
  grayDark,
  white,
  blue,
  grayLight,
  grayDarker,
  redLighter,
  redLightest,
  boxShadow,
} from '../style/colors'
import { fontFamily, fontSize, fontWeight, lineHeight } from '../style/fonts'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
`

interface ITextArea {
  disabled?: boolean
  fullHeight?: boolean
  hasError?: boolean
}

const backgroundCss = ({ disabled, hasError }: ITextArea): string => {
  if (disabled) {
    return grayLight
  }
  if (hasError) {
    return redLightest
  }
  return white
}

const borderCss = ({ disabled, hasError }: ITextArea): string => {
  if (disabled) {
    return 'none'
  }
  if (hasError) {
    return `1px solid ${redDark}`
  }
  return `1px solid ${gray}`
}

export const StyledTextArea = styled.textarea<ITextArea>`
  border-radius: 4px;
  box-shadow: 2px 2px 0 2px transparent;
  transition-property: border-width, border-color, box-shadow;
  transition-duration: 0.1s;
  transition-timing-function: ease-in;
  background-color: ${(props): string =>
    backgroundCss({ disabled: props.disabled, hasError: props.hasError })};
  border: ${(props): string =>
    borderCss({ disabled: props.disabled, hasError: props.hasError })};
  box-sizing: border-box;
  margin-top: 8px;
  margin-bottom: 8px;
  resize: none;
  cursor: ${(props): string => (props.disabled ? 'not-allowed' : '')};
  padding: 8px;
  color: ${({
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    hasError,
  }): string => (hasError ? redDark : grayDarker)};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${fontWeight};
  line-height: ${lineHeight};
  height: ${({
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    fullHeight,
  }): string => (fullHeight ? '100%' : 'auto')};
  :focus {
    border: 1px solid
      ${({
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        hasError,
      }): string => (hasError ? redDark : blue)};
    box-shadow: 0px 0px 0px 3px
      ${({
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        hasError,
      }): string => (hasError ? redLighter : boxShadow)};
    outline: none;
    transition-property: border-width, border-color, box-shadow;
    transition-duration: 0.1s;
    transition-timing-function: ease-in;
  }
  &::placeholder {
    color: ${grayDark};
  }
`

export const StyledLabel = styled.label`
  display: flex;
  flex-direction: column;
`
