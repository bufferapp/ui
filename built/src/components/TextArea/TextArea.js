var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from 'react';
import PropTypes from 'prop-types';
import Text from '../Text';
import { HelpTextWrapper, HelpText } from '../Input/style';
import { Warning } from '../Icon';
import { Container, StyledTextArea } from './style';
export default class TextArea extends React.Component {
    render() {
        const _a = this.props, { value, label, hasError, help, disabled, rows, onChange, id, fullHeight, forwardRef } = _a, props = __rest(_a, ["value", "label", "hasError", "help", "disabled", "rows", "onChange", "id", "fullHeight", "forwardRef"]);
        return (React.createElement(Container, null,
            React.createElement(Text, { htmlFor: id, type: "label" }, label),
            React.createElement(StyledTextArea, Object.assign({ id: id }, props, { hasError: hasError, disabled: disabled, rows: rows > 20 ? 20 : rows, value: value, onChange: onChange, fullHeight: fullHeight, ref: forwardRef })),
            hasError && (React.createElement(HelpTextWrapper, null,
                React.createElement(Warning, { size: "medium" }),
                React.createElement(HelpText, { type: "help", htmlFor: id, hasError: hasError }, help)))));
    }
}
TextArea.propTypes = {
    label: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    hasError: PropTypes.bool,
    disabled: PropTypes.bool,
    onChange: PropTypes.func.isRequired,
    rows: PropTypes.number,
    id: PropTypes.string.isRequired,
    fullHeight: PropTypes.bool,
    forwardRef: PropTypes.oneOfType([
        PropTypes.func,
        PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
    ]),
};
TextArea.defaultProps = {
    placeholder: undefined,
    hasError: false,
    disabled: false,
    rows: 4,
    fullHeight: false,
    forwardRef: undefined,
};
//# sourceMappingURL=TextArea.js.map