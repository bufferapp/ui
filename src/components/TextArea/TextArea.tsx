import React from 'react'
import PropTypes from 'prop-types'
import Text from '../Text'
import { HelpTextWrapper, HelpText } from '../Input/style'
import { Warning } from '../Icon'
import { Container, StyledTextArea } from './style'

export default class TextArea extends React.Component {
  render() {
    const {
      // @ts-expect-error TS(2339) FIXME: Property 'value' does not exist on type 'Readonly<... Remove this comment to see the full error message
      value,
      // @ts-expect-error TS(2339) FIXME: Property 'label' does not exist on type 'Readonly<... Remove this comment to see the full error message
      label,
      // @ts-expect-error TS(2339) FIXME: Property 'hasError' does not exist on type 'Readon... Remove this comment to see the full error message
      hasError,
      // @ts-expect-error TS(2339) FIXME: Property 'help' does not exist on type 'Readonly<{... Remove this comment to see the full error message
      help,
      // @ts-expect-error TS(2339) FIXME: Property 'disabled' does not exist on type 'Readon... Remove this comment to see the full error message
      disabled,
      // @ts-expect-error TS(2339) FIXME: Property 'rows' does not exist on type 'Readonly<{... Remove this comment to see the full error message
      rows,
      // @ts-expect-error TS(2339) FIXME: Property 'onChange' does not exist on type 'Readon... Remove this comment to see the full error message
      onChange,
      // @ts-expect-error TS(2339) FIXME: Property 'id' does not exist on type 'Readonly<{}>... Remove this comment to see the full error message
      id,
      // @ts-expect-error TS(2339) FIXME: Property 'fullHeight' does not exist on type 'Read... Remove this comment to see the full error message
      fullHeight,
      // @ts-expect-error TS(2339) FIXME: Property 'forwardRef' does not exist on type 'Read... Remove this comment to see the full error message
      forwardRef,
      ...props
    } = this.props
    return (
      <Container>
        <Text htmlFor={id} type="label">
          {label}
        </Text>
        <StyledTextArea
          id={id}
          {...props}
          hasError={hasError}
          disabled={disabled}
          rows={rows > 20 ? 20 : rows}
          value={value}
          onChange={onChange}
          fullHeight={fullHeight}
          ref={forwardRef}
        />
        {hasError && (
          <HelpTextWrapper>
            <Warning size="medium" />
            <HelpText type="help" htmlFor={id} hasError={hasError}>
              {help}
            </HelpText>
          </HelpTextWrapper>
        )}
      </Container>
    )
  }
}

// @ts-expect-error TS(2339) FIXME: Property 'propTypes' does not exist on type 'typeo... Remove this comment to see the full error message
TextArea.propTypes = {
  /** It adds a label on top of the textarea box. */
  label: PropTypes.string.isRequired,
  /** It's the placeholder value of the textarea. */
  placeholder: PropTypes.string,
  /** It colors the field in red. */
  hasError: PropTypes.bool,
  /** It disables the textarea field. */
  disabled: PropTypes.bool,
  /** The onChange event */
  onChange: PropTypes.func.isRequired,
  /** Number of rows, max 20 */
  rows: PropTypes.number,
  /** The id to link the textarea with the label */
  id: PropTypes.string.isRequired,
  /** If the textarea should take the height of the parent div */
  fullHeight: PropTypes.bool,
  /**
   * this consumed by the default export that is wrapping the component into a ForwardRef
   * @ignore
   */
  forwardRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
  ]),
}

// @ts-expect-error TS(2339) FIXME: Property 'defaultProps' does not exist on type 'ty... Remove this comment to see the full error message
TextArea.defaultProps = {
  placeholder: undefined,
  hasError: false,
  disabled: false,
  rows: 4,
  fullHeight: false,
  forwardRef: undefined,
}
