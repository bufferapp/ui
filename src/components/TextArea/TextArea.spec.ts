/* eslint-disable react/jsx-filename-extension */
import snap from 'jest-auto-snapshots'
import 'jest-styled-components'

import TextArea from './TextArea'

snap(TextArea, './TextArea.jsx')
