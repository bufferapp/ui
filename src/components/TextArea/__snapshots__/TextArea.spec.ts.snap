// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > TextArea Matches snapshot when boolean prop "disabled" is set to: "false" 1`] = `
<styled.div>
  <Text
    color="grayDarker"
    hasError={false}
    htmlFor="jest-auto-snapshots String Fixture"
    type="label"
  >
    jest-auto-snapshots String Fixture
  </Text>
  <styled.textarea
    disabled={false}
    fullHeight={true}
    hasError={true}
    id="jest-auto-snapshots String Fixture"
    onChange={[MockFunction]}
    placeholder="jest-auto-snapshots String Fixture"
    rows={1}
  />
  <styled.div>
    <WarningIcon
      size="medium"
    />
    <Styled(Text)
      hasError={true}
      htmlFor="jest-auto-snapshots String Fixture"
      type="help"
    />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > TextArea Matches snapshot when boolean prop "fullHeight" is set to: "false" 1`] = `
<styled.div>
  <Text
    color="grayDarker"
    hasError={false}
    htmlFor="jest-auto-snapshots String Fixture"
    type="label"
  >
    jest-auto-snapshots String Fixture
  </Text>
  <styled.textarea
    disabled={true}
    fullHeight={false}
    hasError={true}
    id="jest-auto-snapshots String Fixture"
    onChange={[MockFunction]}
    placeholder="jest-auto-snapshots String Fixture"
    rows={1}
  />
  <styled.div>
    <WarningIcon
      size="medium"
    />
    <Styled(Text)
      hasError={true}
      htmlFor="jest-auto-snapshots String Fixture"
      type="help"
    />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > TextArea Matches snapshot when boolean prop "hasError" is set to: "false" 1`] = `
<styled.div>
  <Text
    color="grayDarker"
    hasError={false}
    htmlFor="jest-auto-snapshots String Fixture"
    type="label"
  >
    jest-auto-snapshots String Fixture
  </Text>
  <styled.textarea
    disabled={true}
    fullHeight={true}
    hasError={false}
    id="jest-auto-snapshots String Fixture"
    onChange={[MockFunction]}
    placeholder="jest-auto-snapshots String Fixture"
    rows={1}
  />
</styled.div>
`;

exports[`jest-auto-snapshots > TextArea Matches snapshot when passed all props 1`] = `
<styled.div>
  <Text
    color="grayDarker"
    hasError={false}
    htmlFor="jest-auto-snapshots String Fixture"
    type="label"
  >
    jest-auto-snapshots String Fixture
  </Text>
  <styled.textarea
    disabled={true}
    fullHeight={true}
    hasError={true}
    id="jest-auto-snapshots String Fixture"
    onChange={[MockFunction]}
    placeholder="jest-auto-snapshots String Fixture"
    rows={1}
  />
  <styled.div>
    <WarningIcon
      size="medium"
    />
    <Styled(Text)
      hasError={true}
      htmlFor="jest-auto-snapshots String Fixture"
      type="help"
    />
  </styled.div>
</styled.div>
`;

exports[`jest-auto-snapshots > TextArea Matches snapshot when passed only required props 1`] = `
<styled.div>
  <Text
    color="grayDarker"
    hasError={false}
    htmlFor="jest-auto-snapshots String Fixture"
    type="label"
  >
    jest-auto-snapshots String Fixture
  </Text>
  <styled.textarea
    disabled={false}
    fullHeight={false}
    hasError={false}
    id="jest-auto-snapshots String Fixture"
    onChange={[MockFunction]}
    rows={4}
  />
</styled.div>
`;
